### ISSUE TITLE
Podspec should not use xcconfig

### ISSUE THREAD

**Issue by friedbunny** [mapbox/mapbox-navigation-ios#152]
Per https://github.com/CocoaPods/CocoaPods/issues/5952, using `xcconfig` in a podspec is deprecated/discouraged and causes the following error in users’ projects:

```
[!] The `ProjectName [Debug]` target overrides the `SWIFT_VERSION` build setting defined in `Pods/Target Support Files/Pods-ProjectName/Pods-ProjectName.debug.xcconfig'. This can lead to problems with the CocoaPods installation 
- Use the `$(inherited)` flag, or 
- Remove the build settings from the target. 
```

If necessary, [`pod_target_xcconfig`](https://guides.cocoapods.org/syntax/podspec.html#pod_target_xcconfig) should be used instead.

/cc @frederoni

**Comment by 1ec5** [comment]
Also https://github.com/Project-OSRM/osrm-text-instructions.swift/blob/master/OSRMTextInstructions.podspec.

**Comment by frederoni** [comment]
Fixed in #156
